staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -164,9 +161,9 @@ type Store struct {
 	// Called to cleanup clients used by the underlying Storage; optional.
 	DestroyFunc func()
 	// Maximum size of the watch history cached in memory, in number of entries.
-	// A zero value here means that a default is used. This value is ignored if
-	// Storage is non-nil.
-	WatchCacheSize int
+	// This value is ignored if Storage is non-nil. Nil is replaced with a default value.
+	// A zero integer will disable caching.
```

Comment:
above you're testing for `> 0`, not `>= 0`, so I think zero currently does `defaultWatchCacheSize`